ucm2/Intel/sof-hda-dsp/HiFi-sof.conf

```diff
@@ -0,0 +1,189 @@
+	      HeadphoneFirBlob "${var:BlobPath}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${var:BlobPath}/drc/passthrough.blob"
+	}
+	False.Define {
+	      SOFIPCVer "ipc3"
+	      BlobPath "${ConfTopDir}/sof/${var:SOFIPCVer}"
+	      PostMixerAnalogPlaybackIIRBytes "EQIIR1.0 eqiir_coef_1"
+	      PostMixerAnalogPlaybackFIRBytes "EQFIR1.0 eqfir_coef_1"
+	      PostMixerAnalogPlaybackDRCBytes "not available"
+	      PostMixerAnalogPlaybackDRCSwitch "not available"
+	      SpeakerIirBlob "${var:BlobPath}/eq_iir/highpass_100hz_0db_48khz.blob"
+	      SpeakerFirBlob "${var:BlobPath}/eq_fir/pass.blob"
+	      SpeakerDrcBlob "${var:BlobPath}/drc/speaker_default.blob"
+	      HeadphoneIirBlob "${var:BlobPath}/eq_iir/pass.blob"
+	      HeadphoneFirBlob "${var:BlobPath}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${var:BlobPath}/drc/passthrough.blob"
+	}
+}
+
+Define.ConfPathFromDMI "${sys:devices/virtual/dmi/id/sys_vendor}/${sys:devices/virtual/dmi/id/product_name}.conf"
+Define.SOFProductConfig "/sof/product_configs/${var:ConfPathFromDMI}"
+Define.SOFConfFullPath "${ConfTopDir}${var:SOFProductConfig}"
+Define.SOFUserConfig "/sof/user_configs/${var:ConfPathFromDMI}"
+Define.SOFUserConfFullPath "${ConfTopDir}${var:SOFUserConfig}"
+
+If.HasUserConfig {
+	Condition {
+		Type Path
+		Mode read
+		Path "${var:SOFUserConfFullPath}"
+	}
+	True {
+		Include.product-config.File "${var:SOFUserConfig}"
+	}
+	False.If.HasProductConfig {
+		Condition {
+			Type Path
+			Mode read
+			Path "${var:SOFConfFullPath}"
+		}
+		True {
+			Include.product-config.File "${var:SOFProductConfig}"
+		}
+	}
+}
+
+# Redefine headphone control FOR SOF. This is same as Headphone in HDA/HiFi-analog.conf
+# but adds DRC and EQ controls.
+
+DefineMacro.SofControl {
+        Define.EndpointDrcVar "${var:__endpoint}DrcBlob"
+        Define.EndpointIirVar "${var:__endpoint}IirBlob"
+        Define.EndpointFirVar "${var:__endpoint}FirBlob"
+        Define.EndpointDrcBlob "${var:$EndpointDrcVar}"
+        Define.EndpointIirBlob "${var:$EndpointIirVar}"
+        Define.EndpointFirBlob "${var:$EndpointFirVar}"
+
+       If.endpoint_with_drc {
+                Condition {
+                        Type ControlExists
+                        Control "name='${var:PostMixerAnalogPlaybackDRCSwitch}'"
+                }
+                True {
+                        EnableSequence [
+                                cset "name='${var:PostMixerAnalogPlaybackDRCSwitch}' ${var:__drcswitch}"
+                                cset-tlv "name='${var:PostMixerAnalogPlaybackDRCBytes}' ${var:EndpointDrcBlob}"
+                                cset-tlv "name='${var:PostMixerAnalogPlaybackIIRBytes}' ${var:EndpointIirBlob}"
+                                cset-tlv "name='${var:PostMixerAnalogPlaybackFIRBytes}' ${var:EndpointFirBlob}"
+                                #shell "/bin/echo '${var:__endpoint} ${var:EndpointIirBlob} ${var:EndpointFirBlob} ${var:EndpointDrcBlob}' >> /tmp/alsa-ucm.txt"
```

Comment:
is this command left intentionally @singalsu?

Reply:
This looks like a debug statement. To raise the log level, uncomment it.

Is there a better logging solution?

Reply:
I've commented it out because the log file write breaks UCM when the user logs in. When the shell command is enabled I need to "sudo chmod a+rw /tmp/alsa-ucm.txt" and restart ucm to get it working and logging the blob data file applies to DSP. A debug print to e.g. /var/log/syslog would be great but I haven't seen such in documentation.